src/AcceptanceTests/TransportEncoding/When_receiving_a_message_with_unknown_transport_encoding.cs

```diff
@@ -27,20 +30,25 @@ public class When_receiving_a_message_with_unknown_transport_encoding : NService
                         return bus.Send(new MyMessage { Id = ctx.OriginalMessageId }, sendOptions);
                     }))
                     .WithEndpoint<Receiver>()
-                    .Done(ctx =>
-                    {
-                        var namespaceManager = new NamespaceManagerAdapter(NamespaceManager.CreateFromConnectionString(Environment.GetEnvironmentVariable("AzureServiceBusTransport.ConnectionString"))); // get connection string from test env
-                        var factory = MessagingFactory.CreateAsync(namespaceManager.Address, namespaceManager.Settings.TokenProvider).GetAwaiter().GetResult();
-                        var queueClient = factory.CreateQueueClient("receivingamessagewithunknowntransportencoding.receiver/$DeadLetterQueue", ReceiveMode.ReceiveAndDelete); // queue name from the test name
-                        var message = queueClient.ReceiveAsync().GetAwaiter().GetResult();
-                        var receivedMessageIdMatchesTheOriginal = message.Properties["NServiceBus.MessageId"].ToString() == ctx.OriginalMessageId;
-                        var testRunIdMatchesTheCurrectTestRun = message.Properties["$AcceptanceTesting.TestRunId"] as string == ctx.TestRunId.ToString();
-                        var deliveredOnceOnly = message.DeliveryCount == 1;
-                        ctx.MessageWasMovedToDlq = receivedMessageIdMatchesTheOriginal && testRunIdMatchesTheCurrectTestRun && deliveredOnceOnly;
+                    .Done(ctx => ctx.MessageWasMovedToDlq)
+                    .Run(new RunSettings { TestExecutionTimeout = TimeSpan.FromMinutes(2) });
+
+            // can't apply a message spy pattern here since message will be always poisonous
+            var rawReceiveTask = Task.Run(async () =>
+            {
+                var connectionString = Environment.GetEnvironmentVariable("AzureServiceBusTransport.ConnectionString");
+                var namespaceManager = new NamespaceManagerAdapter(NamespaceManager.CreateFromConnectionString(connectionString));
+                var factory = MessagingFactory.CreateAsync(namespaceManager.Address, namespaceManager.Settings.TokenProvider).GetAwaiter().GetResult();
+                var dlqPath = Conventions.EndpointNamingConvention(typeof(Receiver)) + "/$DeadLetterQueue";
+                var receiver = await factory.CreateMessageReceiverAsync(dlqPath, ReceiveMode.ReceiveAndDelete);
+                var message = await receiver.ReceiveAsync();
```

Comment:
I could constrain it to the time the test is running. But is it necessary? Wouldn't the task get terminated with test is done?